### PR TITLE
Support D-Bus secret service via libsecret

### PR DESCRIPTION
**Pull request by eagletmt**
- Extract platform-independent API:
  - envchain_search_namespaces
  - envchain_search_values
  - envchain_save_values
- Implement those APIs for Linux using D-Bus secret service

### CODE REVIEW

File: envchain_linux.c [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    callback(key, secret_value_get_text(value), data);
+    secret_value_unref(value);
+    g_hash_table_unref(attrs);
+  }
+  return 0;
+}
+
+void envchain_save_value(const char *name, const char *key, char *value,
+                         G_GNUC_UNUSED int require_passphrase) {
+  /* TODO: Support require_passphrase flag */
```
Comment (eagletmt) [inline envchain_linux.c]: I'm not sure if this is possible or not

**Comment by sorah** [comment]
lgtm?

**Comment by eagletmt** [comment]
I believe it's working well now.
It automatically show prompt requesting to unlock the default collection.

**Comment by sorah** [comment]
LGTM

**Comment by eagletmt** [comment]
squashed

**Pull request merged**